WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs

```diff
@@ -0,0 +1,241 @@
+        }
+
+        [Theory]
+        [MemberData(nameof(Recommendation_request_data))]
+        public void Find_appointments_time_priority(RecommendationRequestDto dto, int availableAppointments)
+        {
+            StubRepositories();
+
+            examinationService.Recommend(dto).Count.ShouldBe(availableAppointments);
+        }
+
+        public static IEnumerable<object[]> Recommendation_request_data()
+        {
+            InitDoctors();
+            InitRooms();
+
+            var retVal = new List<object[]>();
+            DateTime startDate1 = new DateTime(2020, 12, 10, 8, 0, 0);
+            DateTime endDate1 = new DateTime(2020, 12, 10, 16, 0, 0);
+            TimeInterval timeInterval1 = new TimeInterval(startDate1, endDate1);
+
+            DateTime startDate2 = new DateTime(2020, 12, 11, 8, 0, 0);
+            DateTime endDate2 = new DateTime(2020, 12, 11, 16, 0, 0);
+            TimeInterval timeInterval2 = new TimeInterval(startDate2, endDate2);
+
+            DateTime startDate3 = new DateTime(2020, 12, 10, 8, 0, 0);
+            DateTime endDate3 = new DateTime(2020, 12, 10, 10, 0, 0);
+            TimeInterval timeInterval3 = new TimeInterval(startDate3, endDate3);
+
+            timeIntervals.Add(timeInterval1);
+            timeIntervals.Add(timeInterval2);
+            timeIntervals.Add(timeInterval3);
+
+            retVal.Add(new object[] { new RecommendationRequestDto()
+            {
+                DoctorId = doctors[1].Id,
+                TimeInterval = timeIntervals[0],
+                SpecialtyId = 1,
+                Preference = RecommendationPreference.Time
+            }, 5 });          
+
+            retVal.Add(new object[] { new RecommendationRequestDto()
+            {
+                DoctorId = doctors[2].Id,
+                TimeInterval = timeIntervals[1],
+                SpecialtyId = 1,
+                Preference = RecommendationPreference.Time
+            }, 3 });
+
+            retVal.Add(new object[] { new RecommendationRequestDto()
+            {
+                DoctorId = doctors[3].Id,
+                TimeInterval = timeIntervals[2],
+                SpecialtyId = 1,
+                Preference = RecommendationPreference.Time
+            }, 0 });
+
+            retVal.Add(new object[] { new RecommendationRequestDto()
+            {
+                DoctorId = doctors[1].Id,
+                TimeInterval = timeIntervals[0],
+                SpecialtyId = 1,
+                Preference = RecommendationPreference.Doctor
+            }, 5 });
+
+            retVal.Add(new object[] { new RecommendationRequestDto()
+            {
+                DoctorId = doctors[2].Id,
+                TimeInterval = timeIntervals[1],
+                SpecialtyId = 1,
+                Preference = RecommendationPreference.Doctor
+            }, 3 });
```

Comment:
I think it's enough to test case with zero and case with multiple appointments. No need to test three and five available appointments.

Reply:
Five available appointments is maximum amount of displayed appointments. So I tested when there is zero and five available appointments(minimum and maximum). Also I wanted to test if there is number of appointments between minimum and maximum, and because of that I tested case with three appointments.